source/server/http/admin.cc

```diff
@@ -650,7 +665,9 @@ AdminImpl::AdminImpl(const std::string& access_log_path, const std::string& prof
           {"/stats", "print server stats", MAKE_ADMIN_HANDLER(handlerStats), false, false},
           {"/listeners", "print listener addresses", MAKE_ADMIN_HANDLER(handlerListenerInfo), false,
            false},
-          {"/runtime", "print runtime values", MAKE_ADMIN_HANDLER(handlerRuntime), false, false}},
+          {"/runtime", "print runtime values", MAKE_ADMIN_HANDLER(handlerRuntime), false, false},
```

Comment:
My concern with the overrides is that if people are doing local tests, it would be very nice for `/runtime` to be able to print out the original values, and the override value, and which one is in effect (roughly). Is there a way to make that happen? I think that would make this substantially easier to use. In the future you could also imagine something like `POST /runtime_reset` which would reset all changed values to the original (whether on disk or in the base implementation nothing). 

Can you take a look at what would be required to modify `/runtime` to show state in a more obvious way? For `runtime_reset` I would just throw a TODO somewhere.

Reply:
I hear you on that. I'm thinking of formalizing/abstracting the concept of an override and then at the endpoint you can see all three (or more, in the future, who knows) layered - disk, override on disk, override in memory.

My concern is whether it's worth the complexity to make that happen in this PR as opposed to just waiting for more structured admin stuff to land and implementing a better, proto-ized version of this. It would make a good case study in to think about how mutations would interact with structured admin.

So, WDYT about getting this in more-or-less as-is and combining your suggestions with proto-izing these endpoints? It's not about the (mild) engineering effort but more that any complexity might be immediately refactored/deleted anyway.

Reply:
I think we need to fix this as part of this change, even if it's not a super optimal solution. It's going to be too confusing if the user does not know what was overriden using `/runtime_modify`. They will basically have to restart Envoy to get back to a fresh state. We need to be able to view the memory overrides and set them back if desired.

Along these lines, one additional request: Please add a gauge that is something like `runtime.memory_overrides_active` which is set to 1 if any are set, and 0 otherwise. This will allow alarms for when someone changes something if desired.

Reply:
See comment in https://github.com/envoyproxy/envoy/pull/2771 about making this POST. cc @jmarantz